mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -675,12 +676,15 @@ def _local_model_dir(self, source, local_model_path):
                     f"the source artifact location exists and that you can download from "
                     f"it via mlflow.artifacts.download_artifacts()"
                 ) from e
-            # Clean up temporary model directory at end of block. We assume a temporary
-            # model directory was created if the `source` is not a local path (must be downloaded
-            # from remote to a temporary directory)
-            yield local_model_dir
-            if not os.path.exists(source):
-                shutil.rmtree(local_model_dir)
+            try:
+                yield local_model_dir
+            finally:
+                # Clean up temporary model directory at end of block. We assume a temporary
+                # model directory was created if the `source` is not a local path
+                # (must be downloaded from remote to a temporary directory) and
+                # `local_model_dir` is not a UC volumes path
+                if not os.path.exists(source) and not _is_uc_volumes_path(local_model_dir):
```

Comment:
`source` looks like `dbfs:/Volumes/cataglog/schema/my-volume` and `local_model_dir` looks like `/Volumes/cataglog/schema/my-volume`.